core/src/main/java/nl/nn/adapterframework/jdbc/dbms/SqlTranslator.java

```diff
@@ -0,0 +1,162 @@
+/*
+   Copyright 2020 WeAreFrank!
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+package nl.nn.adapterframework.jdbc.dbms;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.net.URL;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.logging.log4j.Logger;
+
+import nl.nn.adapterframework.jdbc.JdbcException;
+import nl.nn.adapterframework.util.ClassUtils;
+import nl.nn.adapterframework.util.LogUtil;
+
+/**
+ * Sql syntax translator to translate queries
+ * for different database management systems (e.g. Oracle to MsSql or PostgreSql to MySql)
+ */
+public class SqlTranslator implements ISqlTranslator {
+	private final Logger log = LogUtil.getLogger(this);
+
+	private static final String PATTERN_FILE = "sql-translate.properties";
+
+	private Map<String,Pattern> sources;
+	private Map<String,String>  targets;
+	
+	private boolean configured=false;
+
+	public SqlTranslator(String source, String target) throws JdbcException {
+		if (StringUtils.isEmpty(source) || StringUtils.isEmpty(target))
+			throw new IllegalArgumentException("Can not translate from [" + source + "] to [" + target + "]");
+		if (source.equalsIgnoreCase(target)) {
+			log.warn("Same source and target for SqlTranslator. Skipping pattern generation.");
+			return;
+		}
+		try {
+			if (!readPatterns(source,target)) {
+				return;
+			}
+		} catch (Exception e) {
+			throw new JdbcException("cannot create SqlTranslator",e);
+		}
+		configured = true;
+	}
+
+	@Override
+	public boolean canConvert(String from, String to) {
```

Comment:
Moet dit niet configured heten? Of anders een check doen om te kijken of er een Match is in de properties file voor de betreffende translatie?